src/parser/link_url.rs

```
@@ -65,112 +71,59 @@ fn is_allowed_generic_scheme(scheme: &str) -> bool {
             Err(nom::Err::Error(CustomError::InvalidLink))
         }
     }
-
+*/
     pub fn parse(input: &str) -> IResult<&str, LinkDestination, CustomError<&str>> {
-        if let Ok((rest, (link, info))) = parse_url(input) {
-            let (hostname, punycode, scheme) = match info {
-                UrlInfo::CommonInternetSchemeURL {
-                    has_puny_code_in_host_name,
-                    hostname,
-                    ascii_hostname,
-                    scheme,
-                } => {
-                    if has_puny_code_in_host_name {
-                        (
-                            Some(hostname),
-                            Some(PunycodeWarning {
-                                original_hostname: hostname.to_owned(),
-                                punycode_encoded_url: link.replacen(hostname, &ascii_hostname, 1),
-                                ascii_hostname,
-                            }),
-                            scheme,
-                        )
-                    } else {
-                        (Some(hostname), None, scheme)
-                    }
-                }
-                UrlInfo::GenericUrl { scheme, .. } => (None, None, scheme),
-            };
-
+        if let Ok((rest, link_destination)) = parse_link(input) {
             Ok((
                 rest,
-                LinkDestination {
-                    target: link,
-                    hostname,
-                    punycode,
-                    scheme,
-                },
+                link_destination 
             ))
         } else {
             Err(nom::Err::Error(CustomError::InvalidLink))
         }
     }
+    
+    pub fn parse_labelled(input: &str) -> IResult<&str, LinkDestination, CustomError<&str>> {
+        let (mut remaining, mut link) = Self::parse(input)?;
+        if let Some(first) = remaining.chars().next() {
+            if matches!(first, ';' | '.' | ',' | ':') {
+                #[allow(clippy::integer_arithmetic)]
```

no do not ignore the error, please use `.get()` instead, there should be no panics possible